### PR TITLE
DR-5961 Refactor build deploy

### PR DESCRIPTION
**Pull request by jope1**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: scripts/workflow/build-service.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export SERVICE="${SERVICE:-""}"
+echo "Building service: ${SERVICE}"
+
+echo "Copy utility code"
+# copy util code but not the test code
+rsync -av --exclude='test/' ./application-utils/* ./application/"${SERVICE}"
+
+cd ./application/"${SERVICE}"
+
+zip -r deployment.zip .
```
Comment (jope1) [inline scripts/workflow/build-service.sh:15]: Exclude tests from the deployment zip.

File: application/healthcare-services-data-manager/test/test_greeting.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import greeting
+
+
```
Comment (jope1) [inline application/healthcare-services-data-manager/test/test_greeting.py:3]: Remove hello world

File: .github/workflows/build-and-deploy-application.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      workspace:
+            description: "The name of the workspace to deploy the application into"
+            required: true
+            type: string
+
+jobs:
+  build-application:
+    name: "Build Application"
+    runs-on: ubuntu-latest
+    environment: dev
```
[the remark this answers is not in the document]
Comment (jope1) [reply .github/workflows/build-and-deploy-application.yaml:20]: Done.

File: .github/workflows/test-build-and-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  deploy-infrastructure-apply:
+    name: "Apply Deploy Infrastructure"
+    needs:
+      [
+        metadata,
+      ]
+    uses: ./.github/workflows/deploy-infrastructure.yaml
+    with:
+      environment: mgmt
```
[the remark this answers is not in the document]
Comment (jope1) [reply .github/workflows/test-build-and-deploy.yaml:22]: Done

File: .github/workflows/test-build-and-deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    name: Build and deploy application
+    needs:
+      [
+        metadata,
+        deploy-infrastructure-apply
+      ]
+    uses: ./.github/workflows/build-and-deploy-application.yaml
+    with:
+      services: "['healthcare-services-data-manager']"
+      workspace: ${{ needs.metadata.outputs.workspace }}
```
[the remark this answers is not in the document]
Comment (jope1) [reply .github/workflows/test-build-and-deploy.yaml:39]: Done

File: application/healthcare-services-data-manager/test/test_greeting.py (hunk 2 above)
Comment (jope1) [reply application/healthcare-services-data-manager/test/test_greeting.py:3]: Done

**Review by timrickwood — APPROVED** [review]

**Review by timrickwood — APPROVED** [review]

**Pull request merged**